packages/shared/src/hooks/useAccountEmailFlow.ts

```diff
@@ -2,44 +2,58 @@ import { useMemo, useState } from 'react';
 import { useMutation, useQuery } from 'react-query';
 import {
   AuthFlow,
+  getKratosFlow,
   initializeKratosFlow,
   submitKratosFlow,
 } from '../lib/kratos';
 import { getErrorMessage, getNodeByKey, getNodeValue } from '../lib/auth';
 import useTimer from './useTimer';
+import { disabledRefetch } from '../lib/func';
 
 interface UseAccountEmail {
   sendEmail: (email: string) => Promise<unknown>;
+  verifyCode: (code: string) => Promise<unknown>;
   resendTimer: number;
   isLoading?: boolean;
   token?: string;
+  flow?: string;
 }
 
 interface UseAccountEmailProps {
+  flow: EmailFlow;
+  flowId?: string;
   isQueryEnabled?: boolean;
-  onSuccess?: () => void;
+  onSuccess?: (email, id) => void;
+  onVerifyCodeSuccess?: () => void;
   onError?: (error: string) => void;
   onTimerFinished?: () => void;
 }
 
 type EmailFlow = AuthFlow.Recovery | AuthFlow.Verification;
 
-function useAccountEmailFlow(
-  flow: EmailFlow,
-  {
-    isQueryEnabled,
-    onSuccess,
-    onError,
-    onTimerFinished,
-  }: UseAccountEmailProps = {},
-): UseAccountEmail {
+function useAccountEmailFlow({
+  flow,
+  flowId,
+  isQueryEnabled,
+  onSuccess,
+  onVerifyCodeSuccess,
+  onError,
+  onTimerFinished,
+}: UseAccountEmailProps): UseAccountEmail {
   const [resentCount, setResentCount] = useState(0);
   const { timer, setTimer, runTimer } = useTimer(onTimerFinished, 0);
 
   const { data: emailFlow } = useQuery(
     [{ type: flow, sentCount: resentCount }],
-    ({ queryKey: [{ type }] }) => initializeKratosFlow(type),
-    { enabled: isQueryEnabled },
+    ({ queryKey: [{ type }] }) =>
+      flowId
+        ? getKratosFlow(AuthFlow.Recovery, flowId)
+        : initializeKratosFlow(type),
+    {
+      ...(flowId
+        ? { ...disabledRefetch, retry: false, enabled: !!flowId }
```

Comment:
No need for the `enabled` now since the starting condition checks if it has a value.
```suggestion
        ? { ...disabledRefetch, retry: false }
```

Reply:
Still would be good to wait for the flow to be available no?
Else this will simply call and fail?

Reply:
Then I think we need to have the `enabled: false` on the `else` value in the ternary operator.

Reply:
That one fires based on `isQueryEnabled`?
I'm just not a 100% sure I'm following your concern here, perhaps you can explain it in our 1:1?

Reply:
Oh, it is just that the value for `enabled` will always be true since in the ternary operator the first object won't be created unless the `flowId` is true. So omitting it should be fine. Sure, let's talk about it in our 1:1 if it is not clear.